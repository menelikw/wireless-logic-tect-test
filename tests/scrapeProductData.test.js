jest.mock('node-fetch')
const fs = require("fs");
const path = require("path");
const fetch = require("node-fetch")

const getProductDataFromPage = require("../scrapeProductData.js").getProductDataFromPage;
const mockProductPageHTML = fs.readFileSync(path.resolve(__dirname, './mock-data/mock-product-page.html'), 'utf8')
const mockEmptyPageHTML = fs.readFileSync(path.resolve(__dirname, 'mock-data/mock-empty-page.html'), 'utf8')

describe('scrapeProductData', () => {
    describe('when page has matching elements', () => {
        it('scrapes product html and transforms it to json ordered by descending price', async () => {
            fetch.mockResolvedValueOnce({text: () => Promise.resolve(mockProductPageHTML)});

            expect(await getProductDataFromPage()).toMatchSnapshot()
        })
    })

    describe('when there are no matching elements', () => {
        it('returns an empty array', async () => {
            fetch.mockResolvedValueOnce({text: () => Promise.resolve(mockEmptyPageHTML)});

            expect(await getProductDataFromPage()).toStrictEqual([])
        })
    })
});