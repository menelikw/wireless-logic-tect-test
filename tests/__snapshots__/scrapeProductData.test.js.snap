// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`scrapeProductData when page has matching elements scrapes product html and transforms it to json ordered by descending price 1`] = `
[
  {
    "description": "Up to 6GB of data per year including 240 SMS (5p / MB data and 4p / SMS thereafter)",
    "discount": 5,
    "optionTitle": "Test package name 1",
    "price": 90,
  },
  {
    "description": "Up to 6GB of data per year including 240 SMS (5p / MB data and 4p / SMS thereafter)",
    "discount": 100,
    "optionTitle": "Test package name 3",
    "price": 80,
  },
  {
    "description": "Up to 6GB of data per year including 240 SMS (5p / MB data and 4p / SMS thereafter)",
    "discount": 10,
    "optionTitle": "Test package name 2",
    "price": 70,
  },
]
`;
